themes/gatsby-theme-blog/src/components/layout.js

```diff
@@ -1,130 +1,124 @@
 import React from "react"
 import { Link } from "gatsby"
+import { Global } from "@emotion/core"
+import { useColorMode, css, Styled, Layout, Header, Container } from "theme-ui"
 import Toggle from "./toggle"
-// import Helmet from "react-helmet";
-import ThemeContext from "../context/theme-context"
 
-import { rhythm, scale } from "../utils/typography"
 import sun from "../../content/assets/sun.png"
 import moon from "../../content/assets/moon.png"
 
-const renderHeader = props => {
+const Title = props => {
   const { location, title } = props
   const rootPath = `${__PATH_PREFIX__}/`
 
   if (location.pathname === rootPath) {
     return (
-      <h1
-        style={{
-          ...scale(0.75),
-          marginBottom: 0,
-          marginTop: 0,
-        }}
+      <Styled.h1
+        css={css({
+          my: 0,
+          fontSize: 4,
+        })}
       >
-        <Link
-          style={{
+        <Styled.a
+          as={Link}
+          css={{
+            color: `inherit`,
             boxShadow: `none`,
             textDecoration: `none`,
-            color: `var(--textTitle)`,
           }}
           to={`/`}
         >
           {title}
-        </Link>
-      </h1>
+        </Styled.a>
+      </Styled.h1>
     )
   } else {
     return (
-      <h3
-        style={{
-          fontFamily: `Montserrat, sans-serif`,
-          marginTop: 0,
-          marginBottom: 0,
-          height: 42, // because
-          lineHeight: `2.625rem`,
-        }}
+      <Styled.h3
+        css={css({
+          my: 0,
+        })}
       >
-        <Link
-          style={{
+        <Styled.a
+          as={Link}
+          css={css({
             boxShadow: `none`,
             textDecoration: `none`,
-            color: `rgb(102, 185, 191)`,
-          }}
+            color: `primary`,
+          })}
           to={`/`}
         >
           {title}
-        </Link>
-      </h3>
+        </Styled.a>
+      </Styled.h3>
     )
   }
 }
 
-const Layout = props => {
+export default props => {
   const { children } = props
+  const [colorMode, setColorMode] = useColorMode()
+  const isDark = colorMode === `dark`
+  const toggleColorMode = e => {
+    setColorMode(isDark ? `light` : `dark`)
+  }
 
   return (
-    <ThemeContext.Consumer>
-      {theme => (
-        <div
-          style={{
-            color: `var(--textNormal)`,
-            background: `var(--bg)`,
-            transition: `color 0.2s ease-out, background 0.2s ease-out`,
-            minHeight: `100vh`,
-          }}
+    <Styled.root>
+      <Layout>
+        <Global
```

Comment:
`Global` will still bleed into shadowed components. For example, if `Layout` has a `Heading`, user shadows `Heading` and applies their own styles and they now have to account for the `Global` styles exerting influence on their `Heading`.

Reply:
Yeah, I'm definitely open to better alternatives for how to handle this. Here, `Global` is being used for the dark and light color modes only. Without setting the body's background color, there will be a visible white area in dark mode when scrolling past the viewport.

There's also an issue where the background color will flash from white to dark when loading a page that has dark mode on. The `useDarkMode` hook [uses a script](https://github.com/donavon/use-dark-mode/blob/develop/noflash.js.txt) to apply classnames to the `body` to avoid this, and I'm guessing we'll want something similar for `theme-ui`, but without using global CSS. We could define "scoped" classnames in a `Global` component for this, but not sure if that will really avoid the issue of styles bleeding into shadowed components. Maybe `theme-ui` can include something like a `Colors` component to handle this in a way that doesn't expose Emotion's `Global` component in Gatsby themes.